tests/test_helper/test_helper.go

```diff
@@ -0,0 +1,234 @@
+	if err := json.NewDecoder(workType).Decode(&workTypes); err != nil {
+		t.Fatal(err)
+	}
+
+	workTypeModels := []entity.WorkType{}
+
+	for _, workType := range workTypes {
+		workTypeModel := entity.WorkType{
+			Name: workType.Name,
+		}
+		workTypeModels = append(workTypeModels, workTypeModel)
+	}
+
+	err = db.Create(&workTypeModels).Error
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func SeedUser(t *testing.T, db *gorm.DB) {
+	makeCircleID := func(s int) *int {
+		return &s
+	}
+
+	userModels := []entity.User{
+		{
+			ID:       1,
+			Name:     "User 1",
+			Email:    "user1@example.com",
+			Hash:     "hash1",
+			CircleID: makeCircleID(1),
+		},
+		{
+			ID:       2,
+			Name:     "User 2",
+			Email:    "user2@example.com",
+			Hash:     "hash2",
+			CircleID: makeCircleID(2),
+		},
+		{
+			ID:       3,
+			Name:     "User 3",
+			Email:    "user3@example.com",
+			Hash:     "hash3",
+			CircleID: makeCircleID(3),
+		},
+	}
+
+	err := db.Create(&userModels).Error
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func GetConnURL(t *testing.T, ctx context.Context) (string, testcontainers.Container) {
+	// Define container request
+	containerReq := testcontainers.ContainerRequest{
+		Image:        "postgres:16",
+		ExposedPorts: []string{"5432/tcp"},
+		Env: map[string]string{
+			"POSTGRES_DB":       "testdb",
+			"POSTGRES_USER":     "user",
+			"POSTGRES_PASSWORD": "foobar",
+		},
+		WaitingFor: wait.ForLog("database system is ready to accept connections").
+			WithOccurrence(2).
+			WithStartupTimeout(5 * time.Second),
+	}
+
+	// Start the PostgreSQL container
+	container, err := testcontainers.GenericContainer(ctx, testcontainers.GenericContainerRequest{
+		ContainerRequest: containerReq,
+		Started:          true,
+	})
+	if err != nil {
+		t.Fatalf("Could not start PostgreSQL container: %s", err)
+	}
+
+	// Get the mapped port for PostgreSQL
+	port, err := container.MappedPort(ctx, "5432")
+	if err != nil {
+		t.Fatalf("Could not get mapped port: %s", err)
+	}
+
+	// Construct the connection URL
+	connURL := fmt.Sprintf("postgres://user:foobar@localhost:%s/testdb?sslmode=disable", port.Port())
+	return connURL, container
+}
+
+//! save old code for now
+// func GetConnURL(t *testing.T, ctx context.Context) (string, testcontainers.Container) {
+// 	container, err := postgres.RunContainer(ctx,
+// 		testcontainers.WithImage("postgres:16"),
+// 		postgres.WithDatabase("testdb"),
+// 		postgres.WithUsername("user"),
+// 		postgres.WithPassword("foobar"),
+// 		testcontainers.WithWaitStrategy(
+// 			wait.ForLog("database system is ready to accept connections").
+// 				WithOccurrence(2).
+// 				WithStartupTimeout(5*time.Second),
+// 		),
+// 	)
+
+// 	if err != nil {
+// 		t.Fatalf("Could not start postgres container: %s", err)
+// 	}
+
+// 	connUrl, err := container.ConnectionString(ctx, "sslmode=disable")
+// 	if err != nil {
+// 		t.Fatalf("Could not get connection string: %s", err)
+// 	}
+
+// 	return connUrl, container
+// }
```

Comment:
```suggestion
func GetConnURL(t *testing.T, ctx context.Context) (string, testcontainers.Container) {
	container, err := postgres.Run(ctx,
		"postgres:16-alpine",
		postgres.WithDatabase("testdb"),
		postgres.WithUsername("user"),
		postgres.WithPassword("foobar"),
		testcontainers.WithWaitStrategy(
			wait.ForLog("database system is ready to accept connections").
				WithOccurrence(2).
				WithStartupTimeout(5*time.Second),
		),
	)

	if err != nil {
		t.Fatalf("Could not start postgres container: %s", err)
	}

	connUrl, err := container.ConnectionString(ctx, "sslmode=disable")
	if err != nil {
		t.Fatalf("Could not get connection string: %s", err)
	}

	return connUrl, container
}
```
pakai ini aja harusnya ga deprecated
https://testcontainers.com/modules/postgresql/